src/bgen/BindingTouch.cs

```diff
@@ -83,14 +79,17 @@ public class BindingTouch : IDisposable {
 	TypeCache? typeCache;
 	public TypeCache TypeCache => typeCache!;
 
+	LibraryManager? libraryManager;
+	public LibraryManager LibraryManager => libraryManager!;
+
 	bool disposedValue;
 	readonly Dictionary<System.Type, Type> ikvm_type_lookup = new Dictionary<System.Type, Type> ();
 	internal Dictionary<System.Type, Type> IKVMTypeLookup {
 		get { return ikvm_type_lookup; }
 	}
 
 	public TargetFramework TargetFramework {
-		get { return target_framework!.Value; }
+		get { return LibraryManager.TargetFramework; }
```

Comment:
You have a possible NRE here. Lets me explain why. If we call this property BEFORE we call GetAttributeLibraryPath, targetFramework is null, but you are ignoring the compiler warning with the !, which is going to make it hard to debug later.

Why not modify the method 
```csharp
public string GetAttributeLibraryPath (string? attributedll, PlatformName currentPlatform)
```
To return a struct with the required values and the pass it to the rest of the code. That way we ensure that we never get to the point of a NRE due to the order in which the methods are called. For example:

```csharp
public record LibraryInfo (string attributeDll, TargetFramework target, bool skipSystemDrawing) {

	public string AttributeDll { get; } = attributeDll;
	public TargetFramework TargetFramework { get; } = target;
    public bool SkipSystemDrawing { get; } = skipSystemDrawing;
}
```

You will have to add SetTargetFramework as a method of the record that takes a string and make the option parsing store a string rather than the actual enum.

The idea behind the changes is that instead of ignoring the nullability of the instance via the ! operator, we write correct code.